Added from_category function

Th3Whit3Wolf opened the pull request:
Takes a category and returns all crates with given category. This should close [#24](https://github.com/theduke/crates_io_api/issues/24)

theduke reviewed (changes requested):
Hey, thanks for the PR.

Two comments:

a) 
I'd prefer a slightly different design: add a `category: Option<String>` field to `ListOptions`.
That way we don't need separate methods.

b) Can you add a simple test exercising the functionality? eg: just fetch a single page and validate that all results have the right category.

Th3Whit3Wolf commented:
Sorry for taking so long to respond, I switched Linux distros from Arch to Nixos. I have a problem....

I made a test, and it failed. I couldn't figure out why. I know this function work because I use it [here](https://github.com/Th3Whit3Wolf/template-benchmarks-rs/blob/auto_readme/src/build.rs) to help generate a readme. So I rewrote the test to print all of the crate that failed and print the total number of crates missed.


```rust
    #[test]
    fn get_from_category_sync() -> Result<(), Error> {
        // Define category
        let category: String = String::from("template-engine");
        // Instantiate the client.
        let client = test_client();
        // Retrieve crates with category.
        let template_engine_crates = client.from_category(Some(category.clone()))?;
        let mut x = 0;
        for c in template_engine_crates {
            match c.categories {
                Some(cat) => {
                    assert!(cat.contains(&category));
                }
                None => {
                    x += 1;
                    println!("Crate: {}  categories{:?}", c.name, c);
                }
            }
        }
        println!("{} Total", x);
        Ok(())
    }
```

And it told me 133(all of them). [In the json response](https://crates.io/api/v1/crates?category=template-engine&page=1&per_page=10&sort=recent-downloads) all of the categories are null.

theduke commented:
Superseded by by #41.
